src/Mod/TechDraw/App/Tag.cpp

```diff
@@ -0,0 +1,93 @@
+ *   Copyright (c) 2025 WandererFan <wandererfan@gmail.com>                *
+ *   Copyright (c) 2025 Benjamin Bræstrup Sayoc <benj5378@outlook.com>     *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+    #include <boost/random.hpp>
+    #include <boost/uuid/uuid_io.hpp>
+    #include <boost/uuid/uuid_generators.hpp>
+    #include <boost/thread/mutex.hpp>
+    #include <boost/thread/lock_guard.hpp>
+#endif
+
+#include <Base/Reader.h>
+#include <Base/Writer.h>
+
+#include "Tag.h"
+
+using namespace TechDraw;
+
+Tag::Tag()
+{
+    createNewTag();
+}
+
+boost::uuids::uuid Tag::getTag() const
+{
+    return tag;
+}
+
+std::string Tag::getTagAsString() const
+{
+    return boost::uuids::to_string(getTag());
+}
+
+void Tag::setTag(const boost::uuids::uuid& newTag)
+{
+    tag = newTag;
+}
+
+void Tag::createNewTag()
+{
+    // Initialize a random number generator, to avoid Valgrind false positives.
+    // The random number generator is not threadsafe so we guard it.  See
+    // https://www.boost.org/doc/libs/1_62_0/libs/uuid/uuid.html#Design%20notes
+    static boost::mt19937 ran;
+    static bool seeded = false;
+    static boost::mutex random_number_mutex;
+
+    boost::lock_guard<boost::mutex> guard(random_number_mutex);
+
+    if (!seeded) {
+        ran.seed(static_cast<unsigned int>(std::time(nullptr)));
+        seeded = true;
+    }
+    static boost::uuids::basic_random_generator<boost::mt19937> gen(&ran);
+
+    tag = gen();
```

Comment:
@WandererFan might need to consider whether there is a better solution to valgrind false positives. Though low priority issue.

Reply:
All that tag related stuff was copy&paste from Sketcher.  The same comment is in Constraint.cpp.

I've never looked into what the false positives are about.

Reply:
@WandererFan, so you never used Valgrind or needed the tags? Should we just delete all the tag implementation completely then?

Reply:
I've run valgrind on fc, but always trying to figure out why hlr was running so long.

The tag is used as a key in places to retrieve cosmetic items.

Reply:
:'(